packages/main/src/TimePicker.ts

```diff
@@ -697,6 +712,34 @@ class TimePicker extends UI5Element implements IFormInputElement {
 		}
 	}
 
+	get valueStateDefaultText(): string | undefined {
+		if (this.valueState === ValueState.None) {
+			return;
+		}
+
+		return this.valueStateTextMappings[this.valueState];
+	}
+
+	get valueStateTextMappings(): ValueStateAnnouncement {
+		return {
+			[ValueState.Positive]: TimePicker.i18nBundle.getText(VALUE_STATE_SUCCESS),
+			[ValueState.Negative]: TimePicker.i18nBundle.getText(VALUE_STATE_ERROR),
+			[ValueState.Critical]: TimePicker.i18nBundle.getText(VALUE_STATE_WARNING),
+			[ValueState.Information]: TimePicker.i18nBundle.getText(VALUE_STATE_INFORMATION),
+		};
+	}
+
+	get shouldDisplayDefaultValueStateMessage(): boolean {
+		return !this.valueStateMessage.length && this.hasValueStateText;
```

Comment:
there is a special function that tells you if a slot will render text that text into account whitespaces comments, etc.. I think it was called `willShowSomething....`